File: app/src/main/java/com/remote/control/allsmarttv/Activities/StartingScreen.java
package com.remote.control.allsmarttv.Activities;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.google.firebase.FirebaseApp;
import com.remote.control.allsmarttv.utils.ir_utils.SupportedClass;
import com.remote.control.allsmarttv.databinding.ActivitySplashBinding;


public class StartingScreen extends AppCompatActivity {

    ActivitySplashBinding splashBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SupportedClass.loadLangLocale(getBaseContext());
        splashBinding = ActivitySplashBinding.inflate(getLayoutInflater());
        setContentView(splashBinding.getRoot());

        FirebaseApp.initializeApp(this);

        new Handler(Looper.getMainLooper()).postDelayed(() -> {

            if (ContextCompat.checkSelfPermission(StartingScreen.this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_DENIED) {
                Intent intent = new Intent(StartingScreen.this, PermActivity.class);
                startActivity(intent);
                finish();

            } else {
                Intent intent = new Intent(StartingScreen.this, FirstActivity.class);
                startActivity(intent);
                finish();
            }

        }, 3000);


    }
}

File: app/src/main/java/com/remote/control/allsmarttv/activitiesUi/StartingScreen.kt
package com.remote.control.allsmarttv.activitiesUi

import android.Manifest
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.remote.control.allsmarttv.utils.ir_utils.SupportedClass
import com.google.firebase.FirebaseApp
import android.os.Looper
import androidx.core.content.ContextCompat
import android.content.pm.PackageManager
import android.content.Intent
import android.os.Handler
import com.remote.control.allsmarttv.appUi.PermActivity
import com.remote.control.allsmarttv.appUi.FirstActivity
import com.remote.control.allsmarttv.databinding.ActivitySplashBinding
import com.remote.control.allsmarttv.utils.Util

class StartingScreen : AppCompatActivity() {

    lateinit var splashBinding: ActivitySplashBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        SupportedClass.loadLangLocale(baseContext)
        splashBinding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(splashBinding.root)

        FirebaseApp.initializeApp(this)

        Handler(Looper.getMainLooper()).postDelayed({

            Util.showBanner = Util.getSharedPreferences(Util.show_banner).toBoolean()
            Util.showInterstitial = Util.getSharedPreferences(Util.show_interstitial).toBoolean()

            if (ContextCompat.checkSelfPermission(
                    this@StartingScreen, Manifest.permission.ACCESS_FINE_LOCATION
                ) == PackageManager.PERMISSION_DENIED
            ) {
                val intent = Intent(this@StartingScreen, PermActivity::class.java)
                startActivity(intent)
                finish()
            } else {
                val intent = Intent(this@StartingScreen, FirstActivity::class.java)
                startActivity(intent)
                finish()
            }

        }, 3000)
    }
}